src/output_image.rs

```diff
@@ -74,17 +76,32 @@ where
             let output_color = C::from(themed_color).to_be_bytes();
             let output_color = output_color.as_ref();
 
-            for p in Rectangle::new(p * pixel_pitch, pixel_size).points() {
-                if let Ok((x, y)) = <(u32, u32)>::try_from(p) {
-                    let start_index = (x + y * self.size.width) as usize * output_color.len();
 
-                    self.data[start_index..start_index + output_color.len()]
-                        .copy_from_slice(output_color)
+            match self.output_settings.pixel_shape {
```

Comment:
This match should be moved outside the `points` loop to remove it from the hot path.